spec/tree_spec.rb

```diff
@@ -1,14 +1,110 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
+describe Tree do
   it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+    expect(described_class.is_a? Class).to eq true
   end
 end
 
-describe 'Fruit' do
+describe AppleTree do
```

Comment:
Take a look at contexts. I would use those to refactor my tests into a story with a clear happy path and not so happy path.

www.betterspecs.org/#contexts